src/theme/components/Layout.tsx

```diff
@@ -47,6 +48,26 @@ const Layout = ({ children, title }: Props): JSX.Element => (
       <GlobalStyle />
       <Helmet title={title} />
       {children}
+      <CookieConsent
+        location="bottom"
+        cookieName={'ga-disable-G-D2D0FR22RN'}
+        enableDeclineButton={true}
+        flipButtons={true}
+        cookieValue={false}
+        declineCookieValue={true}
+        onDecline={() => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          // This function is installed in javascript by the plugin gatsby-plugin-google-gtag
+          gaOptout();
+
+          // Clean the unnecessary cookies
+          Cookies.remove('_ga', { path: '/' });
+          Cookies.remove('_ga_D2D0FR22RN', { path: '/' });
```

Comment:
❓ could we inject the value as it depends on the `Google Analytics` tracking id?
as we may use a different value for preview and production, this will make the cookie removal without change

Reply:
Yes, I did this in the PR for the  preview and production environment

Reply:
Done in #103